WordPress/Classes/ViewRelated/Activity/CalendarViewController.swift

```diff
@@ -0,0 +1,196 @@
+    private var startDateLabel: UILabel!
+    private var separatorDateLabel: UILabel!
+    private var endDateLabel: UILabel!
+
+    private var startDate: Date?
+    private var endDate: Date?
+
+    weak var delegate: CalendarViewControllerDelegate?
+
+    private enum Constants {
+        static let headerPadding: CGFloat = 16
+    }
+
+    override func viewDidLoad() {
+        title = NSLocalizedString("Choose date range", comment: "Title to choose date range in a calendar")
+
+        // Configure Calendar
+        let calendar = Calendar.current
+        self.calendarCollectionView = CalendarCollectionView(calendar: calendar, style: .year)
+        scrollToCurrentDate()
+
+        // Configure headers and add the calendar to the view
+        let header = startEndDateHeader()
+        let stackView = UIStackView(arrangedSubviews: [
+                                            header,
+                                            WeekdaysHeaderView(calendar: calendar),
+                                            calendarCollectionView
+        ])
+        stackView.axis = .vertical
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+        stackView.setCustomSpacing(Constants.headerPadding, after: header)
+        view.addSubview(stackView)
+        view.pinSubviewToAllEdges(stackView, insets: UIEdgeInsets(top: Constants.headerPadding, left: 0, bottom: 0, right: 0))
+        view.backgroundColor = .basicBackground
+
+        setupNavButtons()
+
+        calendarCollectionView.calDataSource.didSelect = { [weak self] startDate, endDate in
+            self?.updateDates(startDate: startDate, endDate: endDate)
+        }
+
+        calendarCollectionView.calDataSource.didDeselectAllDates = { [weak self] in
+            self?.updateDates(startDate: nil, endDate: nil)
+        }
+
+        calendarCollectionView.scrollsToTop = false
+    }
+
+    override func viewWillTransition(to size: CGSize, with coordinator: UIViewControllerTransitionCoordinator) {
+        super.viewWillTransition(to: size, with: coordinator)
+
+        calendarCollectionView.reloadData()
+        scrollToCurrentDate()
+    }
+
+    private func setupNavButtons() {
+        let doneButton = UIBarButtonItem(title: NSLocalizedString("Done", comment: "Label for Done button"), style: .done, target: self, action: #selector(done))
+        navigationItem.setRightBarButton(doneButton, animated: false)
+
+        navigationItem.setLeftBarButton(UIBarButtonItem(barButtonSystemItem: .cancel, target: self, action: #selector(cancel)), animated: false)
+    }
+
+    private func updateDates(startDate: Date?, endDate: Date?) {
+        self.startDate = startDate
+        self.endDate = endDate
+
+        updateLabels()
+    }
+
+    private func updateLabels() {
+        guard let startDate = startDate else {
+            resetLabels()
+            return
+        }
+
+        let formatter = DateFormatter()
+        formatter.dateFormat = "MMM d, yyyy"
```

Comment:
What about using `setLocalizedDateFormatFromTemplate` ?

```suggestion
        formatter.setLocalizedDateFormatFromTemplate("MMM d, yyyy")
```

Reply:
Done!